src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIDockerService.java

```diff
@@ -0,0 +1,74 @@
+package de.tum.in.www1.artemis.service.connectors.localci;
+
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.PullImageResultCallback;
+import com.github.dockerjava.api.exception.BadRequestException;
+import com.github.dockerjava.api.exception.NotFoundException;
+
+import de.tum.in.www1.artemis.exception.LocalCIException;
+
+/**
+ * Service for Docker related operations in local CI
+ */
+
+@Service
+@Profile("localci")
+
+public class LocalCIDockerService {
+
+    private final ReentrantLock lock = new ReentrantLock();
+
+    private final Logger log = LoggerFactory.getLogger(LocalCIDockerService.class);
+
+    private final DockerClient dockerClient;
+
+    public LocalCIDockerService(DockerClient dockerClient) {
+        this.dockerClient = dockerClient;
+    }
+
+    /**
+     * Pulls a docker image if it is not already present on the system
+     * Uses a lock to prevent multiple threads from pulling the same image
+     *
+     * @param imageName the name of the docker image
+     */
+
+    public void pullDockerImage(String imageName) {
```

Comment:
Same here with an extra space.
```suggestion
     */
    public void pullDockerImage(String imageName) {
```